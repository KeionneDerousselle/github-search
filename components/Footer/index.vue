<template>
  <footer
    v-bind="$attrs"
    class="footer">
    <div class="footer__content">
      <span>
        Created with
        <heart-icon class="w-4 inline-flex fill-current text-pink-900" />
        by <a
          id="footer-dev-link"
          href="https://www.keionne.com"
          class="footer__link">Keionne Derousselle</a>.
      </span>
      <span>
        Take a look at the code
        <a
          id="footer-repo-link"
          href="https://www.github.com/KeionneDerousselle/github-search"
          class="footer__link">here</a>!
      </span>
    </div>
  </footer>
</template>
<script>
import HeartIcon from '@/components/Icons/Heart'

export default {
  components: {
    HeartIcon
  }
}
</script>
<style lang="scss">
.footer {
  .footer__content {
    @apply text-base border-t-2 border-gray-200 pt-6;
  }

  .footer__link {
    @apply text-indigo-600 hover:text-indigo-900 border-b border-transparent hover:border-indigo-900;
  }
}
</style>
