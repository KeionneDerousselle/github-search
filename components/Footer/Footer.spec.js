import Footer from '.'

describe('Footer', () => {
  let wrapper

  beforeAll(() => {
    wrapper = shallowPreMocked(Footer)
  })

  afterAll(() => {
    wrapper.destroy()
  })

  it('should render as expected', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should display a link to my website', () => {
    const devLink = wrapper.get('#footer-dev-link')

    expect(devLink.element).toBeVisible()
    expect(devLink.attributes().href).toBe('https://www.keionne.com')
    expect(devLink.text()).toBe('Keionne Derousselle')
  })

  it('should display a link to this project\'s repo', () => {
    const repoLink = wrapper.get('#footer-repo-link')

    expect(repoLink.element).toBeVisible()
    expect(repoLink.attributes().href).toBe('https://www.github.com/KeionneDerousselle/github-search')
    expect(repoLink.text()).toBe('here')
  })
})
