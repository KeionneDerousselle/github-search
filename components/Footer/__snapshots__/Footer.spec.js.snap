// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render as expected 1`] = `
<footer class="footer">
  <div class="footer__content">
    <p>
      Created with
      <heart-icon-stub class="w-5 h-5 inline-flex fill-current text-pink-500"></heart-icon-stub>
      by <a id="footer-dev-link" href="https://www.keionne.com" target="_blank" class="footer__link">
        Keionne Derousselle<span class="sr-only">Opens the developer's website in a new window.</span></a>. Take a look at the code
      <a id="footer-repo-link" href="https://www.github.com/KeionneDerousselle/github-search" target="_blank" class="footer__link">
        here<span class="sr-only">Opens the source code repository for this website in a new window.</span></a>.
    </p>
  </div>
</footer>
`;
